lib/request.js

```diff
@@ -0,0 +1,120 @@
+const https = require('https');
+const http = require('http');
+const url = require('url');
+
+function parseBuffers (value, contentType) {
+  if (Buffer.isBuffer(value)) {
+    var buffer = value;
+    buffer.contentType = contentType || 'application/octet-stream';
+    return buffer;
+  } else if (Array.isArray(value)) {
+    return value.map(v => parseBuffers(v));
+  } else if (value && typeof value === 'object' && Object.keys(value).length === 1 && typeof value._base64 === 'string') {
+    var buffer = Buffer.from(value._base64, 'base64');
+    buffer.contentType = contentType || 'application/octet-stream';
+    return buffer;
+  } else if (value &&typeof value === 'object') {
+    return Object.keys(value).reduce((n, key) => {
+      n[key] = parseBuffers(value[key]);
+      return n;
+    }, {});
+  } else {
+    return value;
+  }
+};
+
+function followRedirect(cfg, response, callback) {
+  if (cfg.maxRedirects < 1) {
+    throw new Error('Too many redirects');
+  }
+
+  cfg.maxRedirects -= 1;
+
+  let parsedUrl = response.headers.location && url.parse(response.headers.location);
+
+  Object.assign(cfg, {host: parsedUrl.host, path: parsedUrl.pathname, port: parsedUrl.protocol.includes('https') ? 443 : 80});
```

Comment:
We do not use `Object.assign` because it's burned us a few times, use a `.reduce` here to combine into a new object